api/PayrollProcessor.Functions.Api/Features/Departments/DepartmentEmployeeTrigger.cs

```diff
@@ -60,37 +62,48 @@ public class DepartmentEmployeeTrigger
                     return;
 
                 default:
+                    log.LogWarning("Queue message with {eventName} cannot be handled", eventName);
+
                     return;
             }
         }
 
-        private Task HandleEmployeeCreation(CloudQueueMessage queueMessage, ILogger log)
-        {
-            var message = QueueMessageHandler.FromQueueMessage<EmployeeCreation>(queueMessage);
-
-            return queryDispatcher.Dispatch(new EmployeeQuery(message.EmployeeId))
-                .Bind(employee => commandDispatcher.Dispatch(new DepartmentEmployeeCreateCommand(employee, idGenerator.Generate())))
-                .Bind(departmentEmployee => apiClient.SendNotification(nameof(EmployeeUpdatesQueue), departmentEmployee))
-                .Match(
-                    _ => log.LogInformation(""),
-                    () => log.LogError(""),
-                    e => log.LogError(e, ""));
-        }
+        private Task HandleEmployeeCreation(CloudQueueMessage queueMessage, ILogger log) =>
+            queueMessage
+                .Apply(QueueMessageHandler.FromQueueMessage<EmployeeCreation>)
```

Comment:
By using `.Apply` we can move from imperative statement based function execution to a chainable simulation in C# of piping from functional languages.